types/src/transaction/authenticator.rs

```diff
@@ -95,6 +99,11 @@ impl TransactionAuthenticator {
                 secondary_signer_addresses,
                 secondary_signers,
             } => {
+                let num_sigs: usize = sender.number_of_signatures() +
+                    secondary_signers.iter().map(|auth| auth.number_of_signatures()).sum::<usize>();
+                if num_sigs > MAX_NUM_OF_SIGS {
+                    return Err(Error::new(CryptoMaterialError::WrongLengthError));
```

Comment:
I think we probably want to introduce a new error code here (`MaxSigsExceeded`?). I believe `WrongLengthError` is for (e.g.) when a user attempts to use a public key with too few or too many bytes.